### PR TITLE
Update README.md with additional clarifications of instructions

### PR DESCRIPTION
**Pull request by alberttwong**
more troubleshooting info.  more info on why picking AWS regions are important.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -105,6 +105,10 @@ git checkout openshift-ansible-3.0.47-6
 ```
 
 ## Usage
+
+### Pick your AWS Region
+You have to pick which region to deploy your OpenShift installation.   Specify or modify 'run.py' with the correct AWS region and corresponding AMI. 
+
```
Comment (thoraxe) [inline README.md:111]: You don't have to pick a region. You could use the default. Do **NOT** modify run.py.

Please refactor.

File: README.md [hunk 2]
```diff
@@ -114,7 +118,7 @@ export AWS_SECRET_ACCESS_KEY=bar
 ```
 
 ### Add the SSH Key to the SSH Agent
-You will need to add the private key to your SSH agent: 
+Create your EC2 Key Pair with name "default" in the AWS Region you picked.  You will need to add the private key to your SSH agent:
```
Comment (thoraxe) [inline README.md:121]: The keypair does not have to be default. You can specify the keypair to use. Please refactor.

File: README.md [hunk 3]
```diff
@@ -137,8 +141,8 @@ should be used:
     | AMI           | Amazon Region  |
     | ---           | ---            |
     | ami-2051294a* | us-east-1*     |
-    | ami-d1315fb1  | us-west-1      |
-    | ami-775e4f16  | us-west-2      |
+    | ami-d1315fb1  | us-west-1      | N. California
+    | ami-775e4f16  | us-west-2      | Oregon
```
Comment (thoraxe) [inline README.md:145]: Why do you call out the names for these regions but no other regions?

File: README.md [hunk 4]
```diff
@@ -166,6 +170,11 @@ the cluster id, and keypair:
 --r53-zone my.hosted.domain --rhsm-user my_redhat_user --rhsm-pass my_redhat_pass
 ```
 
+More complex version
+```
+./run.py --no-confirm --cluster-id openshift --num-nodes 1 --keypair default --r53-zone alberttwong.com --rhsm-user alwong@redhat.com --rhsm-pass n0n34u --region us-west-1 --ami ami-d1315fb1 --run-smoke-tests --num-smoke-test-users 5 
+```
```
Comment (thoraxe) [inline README.md:176]: Why is this necessary? You are effectively duplicating the --help option with an example.

File: README.md [hunk 5]
```diff
@@ -210,3 +219,6 @@ un-listed error, simply execute `run.py` again with the same exact options.
 #### Cloudformation
 On occasion the Cloudformation stack will fail to delete properly. Simply go
 into the AWS console and find the Cloudformation stack and delete it.
+
+#### When all hope is lost
+Most likely it's because you have existing settings in your AWS region.  Pick another AWS region and start over.
```
Comment (thoraxe) [inline README.md:224]: This is not specific enough and should probably be in the troubleshooting section.

File: README.md (hunk 1 above)
Comment (alberttwong) [inline README.md:111]: You don't but then you need to push instructions that  you need to create the ec2 key in a SPECIFIC region for it to work.   I was trying to allow others to pick their region since not all of us are physically located in the default east region.

File: README.md (hunk 2 above)
Comment (alberttwong) [inline README.md:121]: So you find out about the "default" later in the instructions.   If you picked a different name and then you read the rest of the instructions and tried to use the default run.py, you'd have to go back to this step.   I'm trying to get the happy path and defaults to work.

File: README.md (hunk 3 above)
Comment (alberttwong) [inline README.md:145]: So here's another problem.... you say us-west-1 but in the AMS console, they don't call it that.   So I worked backwards from this instructions and found out what is the name.   I don't know what they call it in the east or other regions because I didn't try to make it work there.

File: README.md (hunk 4 above)
Comment (alberttwong) [inline README.md:176]: It is... because it's the most practical command to create the environment. Let's say if your example was great, I wouldn't need to write this.

File: README.md (hunk 5 above)
Comment (alberttwong) [inline README.md:224]: I thought this was the troubleshooting section.   Also, I couldn't get it to work in N. California AWS.   After trying to get it to work, I gave up.   I just picked another region and it started to work.   People need to know this or else you and your team is going to do support for all of us.

File: README.md (hunk 1 above)
Comment (thoraxe) [inline README.md:111]: OK -- so then the docs should/can mention that the key needs to be in the region. Please refactor.

File: README.md (hunk 2 above)
Comment (thoraxe) [inline README.md:121]: I'm happy for a larger refactoring of getting the defaults to work. Perhaps the prereqs need to be revisited.

File: README.md (hunk 3 above)
Comment (thoraxe) [inline README.md:145]: OK. So please find out the "public names" for the different regions. Thanks!

File: README.md (hunk 4 above)
Comment (thoraxe) [inline README.md:176]: - What's wrong with the example?
- Running `run.py` with no options will prompt you for everything you need
- If the defaults are not sensible, then please submit a PR / suggest changes against the defaults.

Your "more complex version" is only required if you don't want to accept any defaults. Not specifying a user/password will force you to be prompted. It's not required. Smoke tests are optional. Region and AMI are optional unless you don't want to use a default, but they are documented, etc etc.

I'm all for accepting changes but if you are going to use a "more complex version" then you would likely want to include something like all sensible options, not just the few sensible options that might make sense outside of the defaults, no?

File: README.md (hunk 5 above)
Comment (thoraxe) [inline README.md:224]: What issues did you have? Did you file issues? How are we supposed to "get it to work" if we don't know what broke? The statement includes an assumption ("you have existing settings") but doesn't explain how to rectify them. That's not really a troubleshooting maneuver.

What if the settings were the same in _every_ region? THis troubleshooting step wouldn't work.

Sometimes AWS is broken. I've seen it happen.
